File: Solutions_Easy/0020_ValidParentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> stack;

        char temp = ' ';
        for(int i = 0; i < s.size(); i++) {
            if(s[i] == '(' || s[i] == '[' || s[i] == '{') {
                stack.push(s[i]);
            } else {
                if(stack.empty())
                    return false;
                else {
                    temp = stack.top();
                    if(s[i] == ')' && temp != '(' || s[i] == ']' && temp != '[' || s[i] == '}' && temp != '{')
                        return false;
                    stack.pop();
                }
            }
        }
        
        return stack.empty();
    }
};

File: Solutions_Easy/0020_ValidParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        close_to_open = {")": "(", "]": "[", "}": "{"}

        for c in s:
            if c in close_to_open:
                if len(stack) == 0 or stack.pop() != close_to_open[c]:
                    return False
            else:
                stack.append(c)

        return len(stack) == 0
